server/utils/courseParticipationUtils.ts

```diff
@@ -152,80 +153,64 @@ export default class CourseParticipationUtils {
   private static summaryListRows(
     courseParticipationWithName: CourseParticipationWithName,
   ): Array<GovukFrontendSummaryListRowWithValue> {
-    const summaryListRows: Array<GovukFrontendSummaryListRowWithValue> = []
-
-    summaryListRows.push({
-      key: { text: 'Programme name' },
-      value: { text: courseParticipationWithName.name },
-    })
-
-    if (courseParticipationWithName.setting) {
-      const valueTextItems: Array<string> = []
-
-      if (courseParticipationWithName.setting.type) {
-        valueTextItems.push(StringUtils.properCase(courseParticipationWithName.setting.type))
-      }
+    return [
+      CourseParticipationUtils.summaryListRow('Programme name', [courseParticipationWithName.name]),
+      CourseParticipationUtils.summaryListRowSetting(courseParticipationWithName),
+      CourseParticipationUtils.summaryListRowOutcome(courseParticipationWithName),
+      CourseParticipationUtils.summaryListRow('Additional detail', [courseParticipationWithName.outcome.detail]),
+      CourseParticipationUtils.summaryListRow('Source of information', [courseParticipationWithName.source]),
+      CourseParticipationUtils.summaryListRow('Added by', [
+        courseParticipationWithName.addedBy,
+        DateUtils.govukFormattedFullDateString(courseParticipationWithName.createdAt),
+      ]),
+    ]
+  }
 
-      if (courseParticipationWithName.setting.location) {
-        valueTextItems.push(`${courseParticipationWithName.setting.location}`)
-      }
+  // eslint-disable-next-line @typescript-eslint/member-ordering
+  private static summaryListRow(
+    keyText: GovukFrontendSummaryListRowKey['text'],
+    valueTextItems: Array<string | undefined>,
+  ): GovukFrontendSummaryListRowWithValue {
+    return {
+      key: { text: keyText },
+      value: { text: valueTextItems.join(', ') || 'Not known' },
```

Comment:
If there's a `undefined` in here it will still add the `', '`, so you might want to filter out the `undefined` values before this

It doesn't look like any `undefined` values would make it here in an array of more than one item in the current state, but it might be good to tighten this in case things change

Reply:
Good point, I've tightened this up, a bit of a hard one to test as we have no arrays that could possibly have more than one undefined element here (and it's a private method), so haven't updated tests for this.